.github/workflows/upload-to-pypi.yml

```diff
@@ -0,0 +1,110 @@
+      uses: actions/setup-python@v2
+      with:
+        python-version: '3.8'
+    - name: Install deps
+      run: python -m pip install build twine
+    - name: Build SDist
+      run: python -m build --sdist
+    - uses: actions/upload-artifact@v2
+      with:
+        path: dist/*.tar.gz
+    - name: Check metadata
+      run: twine check dist/*
+
+  build_wheels:
+    name: Build wheels on ${{ matrix.os }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest, windows-latest, macos-latest]
+    steps:
+      - uses: actions/checkout@v2
+      # Used to host cibuildwheel
+      - uses: actions/setup-python@v2
+      - name: Install cibuildwheel
+        run: python -m pip install cibuildwheel
+      - name: Build wheels
+        run: python -m cibuildwheel --output-dir wheelhouse
+        env:
+          # Disable explicitly python 3.10 and building PyPy wheels
+          CIBW_SKIP: cp310-* pp*
+          CIBW_PRERELEASE_PYTHONS: False
+          # Manually force a version (and avoid building local wheels)
+          CIBW_ENVIRONMENT: "SETUPTOOLS_SCM_PRETEND_VERSION=${{ github.event.inputs.overrideVersion }}"
+      - uses: actions/upload-artifact@v2
+        with:
+          path: wheelhouse/*.whl
+
+  build_aarch64_wheels:
+    name: Build wheels manylinux_aarch64
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python: [36, 37, 38, 39]
+        include:
+          - os: ubuntu-latest
+            arch: aarch64
+            platform_id: manylinux_aarch64
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v1
+      - name: Install cibuildwheel
+        run: python -m pip install cibuildwheel
+      - name: Build wheels
+        run: python -m cibuildwheel --output-dir wheelhouse
+        env:
+          CIBW_ARCHS_LINUX: ${{matrix.arch}}
+          CIBW_BUILD: cp${{ matrix.python }}-${{ matrix.platform_id }}
+          # Manually force a version (and avoid building local wheels)
+          CIBW_ENVIRONMENT: "SETUPTOOLS_SCM_PRETEND_VERSION=${{ github.event.inputs.overrideVersion }}"
+      - uses: actions/upload-artifact@v2
+        with:
+          path: wheelhouse/*.whl
+
+  upload_all:
+    needs: [build_wheels, build_aarch64_wheels, make_sdist]
+    runs-on: ubuntu-latest
+    steps:
+    - name: Download artifacts
+      uses: actions/download-artifact@v2
+      with:
+        name: artifact
+        path: dist
+    - name: Publish distribution to Test PyPI
+      uses: pypa/gh-action-pypi-publish@v1.4.2
+      with:
+        password: ${{ secrets.TEST_PYPI_API_TOKEN }}
+        repository_url: https://test.pypi.org/legacy/
+        # do not fail if the TestPyPi release already
+        # exists
+        skip_existing: true
+    - name: Publish distribution to PyPI
+      uses: pypa/gh-action-pypi-publish@v1.4.2
+      with:
+        user: __token__
+        password: ${{ secrets.PYPI_API_TOKEN }}
```

Comment:
Not sure why we push to both test pypi and pypi here?
Also what is the token in `user`?

Reply:
The push to test pypi is merely a check that everything goes well before pushing to pypi. It can be removed.

>  Also what is the token in user?

This is the way this action is used in the documentation (see [here](https://github.com/pypa/gh-action-pypi-publish#usage)). It relies on the new API of pypi to authenticate (https://pypi.org/help/#apitoken).

Reply:
Ok I get it for the `__token__`, this is merely to notify that the authentification does not require a user.

I think I would prefer that we don't push to test.pypi, it feels not really useful.